test/testlib/nuodb_database_utilities.go

```diff
@@ -820,13 +820,27 @@ type BackupHookResponse struct {
 	Message string `json:"message"`
 }
 
-func GetBackupHookResponse(t *testing.T, namespaceName, smPod, resourcePath string) *BackupHookResponse {
+func GetBackupHookResponse(t *testing.T, namespaceName, smPod, resourcePath string, payload ...string) *BackupHookResponse {
+	cmdArgs := []string{
+		"exec", smPod, "-c", "engine", "--",
+		"curl", "-s", "-X", "POST", "http://0.0.0.0:8000/" + resourcePath,
+	}
+	// Prepare the payload as map
+	if len(payload) > 0 {
+		require.Equal(t, 0, len(payload)%2, "Backup hook payload must contain even number of entries")
+		payloadMap := make(map[string]string, len(payload)/2)
+		for i := 0; i < len(payload)-1; i++ {
```

Comment:
Shouldn't `i` get incremented by 2?

Reply:
If this will make it read better, I can change it.